etc/spack/defaults/packages.yaml

```diff
@@ -25,9 +25,13 @@ packages:
       fftw-api: [fftw, amdfftw]
       flame: [libflame, amdlibflame]
       fuse: [libfuse]
-      gl: [mesa+opengl, mesa18, opengl]
       glu: [mesa-glu, openglu]
-      glx: [mesa+glx, mesa18+glx, opengl]
+      gl: [libglvnd-fe, mesa, mesa18, opengl]
```

Comment:
You added `provides('gl', when='~glvnd')` to the `opengl` package: I think it would make sense to say:

```
gl: [libglvnd-fe, mesa, mesa18, opengl~glvnd]
```

or did you try that and encounter a problem with it?

Reply:
With the new concretizer,  we don't need to specify the variants in the default packages. If we wanted to be clearer,`gl` should say: 

`gl: [libglvnd-fe, mesa~glvnd, mesa18, opengl~glvnd]`

and the `glx` and `egl` packages should have similar modifications. We decided to let the concretizer be as flexible as possible and removed the variants from default packages.

Reply:
That makes sense: it's true that the concretizer can handle this since you've encoded that information in the `package.py`. IMO it would be easier to read if also updated here (although it's definitely not necessary).